pkg/apis/pipeline/v1/pipeline_validation.go

```diff
@@ -497,6 +497,21 @@ func (pt *PipelineTask) extractAllParams() Params {
 	return allParams
 }
 
+func (pt *PipelineTask) GetVarSubstitutionExpressions() ([]string, bool) {
```

Comment:
Shall we add some more docString for this since it is an exported func?

Reply:
added
```
GetVarSubstitutionExpressions extract all values between the parameters "$(" and ")" of steps and sidecars
```